src/ds3.c

```diff
@@ -1766,10 +1777,27 @@ ds3_error* ds3_get_bucket(const ds3_client* client, const ds3_request* request,
     return NULL;
 }
 
+static int num_chars_in_ds3_str( ds3_str* str, char ch) {
```

Comment:
You have an extra space between the `(` and `ds3_str`.  Also, can ds3_str be `const` here?